.github/workflows/ci.yml

```diff
@@ -27,26 +27,44 @@ jobs:
   lingua-franca:
     runs-on: ubuntu-latest
     steps:
+      - name: Check out reactor-ts repository
+        uses: actions/checkout@v3
+        with: 
+          path: "reactor-ts"
+          fetch-depth: 0
       - name: Read lingua-franca-ref.txt
-        run: echo "lingua-franca-ref=$(wget -qO- https://raw.githubusercontent.com/lf-lang/reactor-ts/${{ github.head_ref || github.ref_name }}/lingua-franca-ref.txt)" >> $GITHUB_ENV
+        run: echo "lingua-franca-ref=$(cat ./reactor-ts/lingua-franca-ref.txt)" >> $GITHUB_ENV
       - name: Check out lingua-franca repository
         uses: actions/checkout@v3
         with:
           repository: lf-lang/lingua-franca
           ref: ${{ env.lingua-franca-ref }}
           fetch-depth: 0
+          path: "lingua-franca"
           submodules: true
+      - name: Move Lingua Franca to default pwd
+        run: |
+          shopt -s dotglob
+          mv -- ./lingua-franca/* ${{ github.workspace }}
```

Comment:
I believe that if you do the checkout of `reactor-ts` without a `path` parameter, it will end up in `${{ github.workspace }}`, and subsequently doing the cloning LF with `path: "lingua-franca"` would put the checkout exactly where you want it? I'm not sure though, but I think all of this should be possible without moving stuff.

Reply:
Here are things we want to do:
1. Put `lingua-franca` in `pwd`
2. Put `reactor-ts` in a random directory which we know, other than `pwd` because `lingua-franca` resides there.

Currently, the following procedure is performed:
1. Checkout `reactor-ts` at `pwd/reactor-ts`
2. Checkout `lingua-franca` at `pwd/lingua-franca`
3. Move everything from `pwd/lingua-franca` to `pwd`.

I tried some simplification and they don't work:
1. Not moving `lingua-franca` to `pwd`. [https://github.com/lf-lang/reactor-ts/pull/157#issuecomment-1566514249](This) is the issue.
2. Checkout `reactor-ts` at `pwd/reactor-ts`, then checkout `lingua-franca` at `pwd`. This doesn't work because `actions/checkout@v3` will purge everything in the directory before cloning, so checkout `lingua-franca` at `pwd` will effectively remove everything in `pwd/reactor-ts`.

Reply:
OK, let's leave as is because it works 😄